pallets/pool-registry/src/lib.rs

```diff
@@ -0,0 +1,373 @@
+			// admin as our depositor.
+			let depositor = ensure_signed(origin).unwrap_or(admin.clone());
+
+			T::ModifyPool::create(
+				admin.clone(),
+				depositor,
+				pool_id,
+				tranche_inputs,
+				currency,
+				max_reserve,
+				metadata,
+			)
+			.map(|_| Self::deposit_event(Event::Created { pool_id, admin }))
+		}
+
+		/// Update per-pool configuration settings.
+		///
+		/// This updates the tranches of the pool,
+		/// sets the minimum epoch length, and maximum NAV age.
+		///
+		/// If no delay is required for updates and redemptions
+		/// don't have to be fulfilled, then this is executed
+		/// immediately. Otherwise, the update is scheduled
+		/// to be executed later.
+		///
+		/// The caller must have the `PoolAdmin` role in order to
+		/// invoke this extrinsic.
+		#[pallet::weight(T::WeightInfo::update_no_execution(T::MaxTranches::get())
+		.max(T::WeightInfo::update_and_execute(T::MaxTranches::get())))]
+		pub fn update(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			changes: PoolChanges<
+				T::Rate,
+				T::MaxTokenNameLength,
+				T::MaxTokenSymbolLength,
+				T::MaxTranches,
+			>,
+		) -> DispatchResultWithPostInfo {
+			// Make sure the following are true for a valid Pool Update
+			// 1. Make sure Origin is signed
+			// 2. Ensure the signed origin is PoolAdmin
+			// 3. Either no changes in tranches/metadata or changes in both
+			// 4. Ensure not in Submission Epoch Time
+			// 5. If no changes, do no_op
+			// 6. Minimum epoch time has to be between min and max epoch time
+			// 7. MaxNavAge has to be under the upper bound NavAge
+			// 8. IsValidTrancheChange?
+
+			let who = ensure_signed(origin)?;
+			ensure!(
+				T::Permission::has(
+					PermissionScope::Pool(pool_id),
+					who,
+					Role::PoolRole(PoolRole::PoolAdmin)
+				),
+				BadOrigin
+			);
+
+			match T::ModifyPool::update(pool_id, changes) {
+				Ok((state, dispatch_info)) => {
+					Self::deposit_event(Event::UpdateRegistered { pool_id });
+
+					match state {
+						UpdateState::NoExecution => (),
+						UpdateState::Executed => {
+							Self::deposit_event(Event::UpdateExecuted { pool_id })
+						}
+						UpdateState::Stored => Self::deposit_event(Event::UpdateStored { pool_id }),
+					}
+
+					Ok(dispatch_info)
+				}
+				Err(e) => Err(e),
+			}
+		}
+
+		/// Executed a scheduled update to the pool.
+		///
+		/// This checks if the scheduled time is in the past
+		/// and, if required, if there are no outstanding
+		/// redeem orders. If both apply, then the scheduled
+		/// changes are applied.
+		#[pallet::weight(T::WeightInfo::execute_update(T::MaxTranches::get()))]
+		pub fn execute_update(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+		) -> DispatchResultWithPostInfo {
+			ensure_signed(origin)?;
+
+			match T::ModifyPool::execute_update(pool_id) {
+				Ok(res) => {
+					Self::deposit_event(Event::UpdateExecuted { pool_id });
+					Ok(res)
+				}
+				Err(e) => Err(e),
+			}
```

Comment:
```rust
T::ModifyPool::execute_update(pool_id).map(|res| {
   Self::deposit_event(Event::UpdateExecuted { pool_id });
   res
})
```

Reply:
We really want `inspect`, but alas it's not stable yet https://github.com/rust-lang/rust/issues/91345